e2e/test/scenarios/dashboard/reproductions/12926-editing-dashboards-doesnt-cancel-running-queries.cy.spec.js

```diff
@@ -47,30 +65,73 @@ describe("issue 12926", () => {
 
       removeCard();
 
-      restoreQuery();
+      restoreDashcardQuery();
 
       undo();
 
-      cy.wait("@cardQueryRestored");
+      cy.wait("@dashcardQueryRestored");
 
-      getDashboardCard().findByText("42");
+      getDashboardCard().findByText(queryResult);
+    });
+  });
+
+  describe("saving a dashboard that retriggers a non saved query (negative id)", () => {
+    it("should stop the ongoing query", () => {
+      // this test requires the card to be manually added to the dashboard, as it requires the dashcard id to be negative
+      cy.createNativeQuestion(questionDetails);
+
+      cy.createDashboard().then(({ body: { id: dashboardId } }) => {
+        cy.visit(`/dashboard/${dashboardId}`);
+      });
```

Comment:
We have `visitDashboard` util, which makes sure the dashboard is loaded before performing the next actions which we use throughout the app. And has been used throughout our tests, so it's more established.

https://github.com/metabase/metabase/blob/6f7f02a9fbad06ccc68826d7aa5c91afcdec1a36/e2e/support/helpers/e2e-misc-helpers.js#L159